### PR TITLE
[master] Fix #2828 Bookmarkable link does not select compound attributes

### LINKED ISSUE #2828: Bookmarkable link does not select compound attributes
A bookmarkable link does not select compound attributes that are specified in the link.

How to reproduce:
- Use an entity that has compound attributes
- Go to the data explorer and select the entity
- Select one of the compound attributes
- Copy the link to the clipboard
- Open a new browser window
- Paste the link in the address bar

Actual result:
The data explorer opens with the selected entity, but the compound attribute is not selected

Expected result:
The data explorer opens with the selected entity and the compound attribute is selected

### PR DESCRIPTION
**Pull request by Mark-de-Haan**
## Note
Tried implementing v2 of the api for getting entityMetaData in the dataexplorer in the tree. This lead to a bug fix in the AttributeMetaDataResponseV2, and adding a method to get attributes in the RestControllerV2.

I reverted the v2 back to v1 for the dataexplorer and the tree because the JSON messages between the two api versions are not comparable, leading to length of undefined errors etc..

See issue #3793

## Note 2
This fix allows for the selecting of compound attributes in the URL.
However, the current tree implementation only allows for selection of first level nodes, so atomic attributes can not be selected via the URL

### CODE REVIEW

**Comment by jjettenn** [comment]
I cannot accept this fix because it is not fixing the real problem. The current tree implementation should be changed.
1. The javascript var setting must contain also the xcompound_string attribute. It exists in the url.
2. The current implementation must, when looking if an attribute is visible, also look al te names of the compound attribute children.

**Comment by Mark-de-Haan** [comment]
Then we should create a story for it because it is much more then just a bug fix

**Comment by jjettenn** [comment]
I have opened this PR because the changes are a step forwards. I am missing some tests: retrieveEntityAttributeMetaPost --> RestControllerV2
retrieveEntityAttributeMeta --> RestControllerV2
concatMetaEntityHrefV2 --> Href

Still I think that a part of the issue is not yet finished and that we cannot close the issue. We should introduce a new issue with the steps we still need to do.

**Comment by Mark-de-Haan** [comment]
Il start writing the tests

On 8 October 2015 at 12:16, jjettenn <notifications@github.com> wrote:

> I have opened this PR because the changes are a step forwards. I am
> missing some tests: retrieveEntityAttributeMetaPost --> RestControllerV2
> retrieveEntityAttributeMeta --> RestControllerV2
> concatMetaEntityHrefV2 --> Href
>
> Still I think that a part of the issue is not yet finished and that we
> cannot close the issue. We should introduce a new issue with the steps we
> still need to do.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/molgenis/molgenis/pull/3794#issuecomment-146485553>.
>



-- 
Mark de Haan
Genomics Coordination Centre
Genetics Department
University Medical Centre Groningen

**Comment by jjettenn** [comment]
Great then we can merge this PR.

**Comment by Mark-de-Haan** [comment]
@jjettenn can you take a look?

**Comment by jjettenn** [comment]
It says that there are some conflicts in the code, can you look into it?

File: molgenis-dataexplorer/src/main/resources/js/dataexplorer.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ function($, molgenis, settingsXhr) {
     self.getIdentifierAttribute = getIdentifierAttribute;
     self.getPatientAttribute = getPatientAttribute;
 
-    var restApi = new molgenis.RestClient();
+    var api = new molgenis.RestClient();
```
Comment (bartcharbon) [inline molgenis-dataexplorer/src/main/resources/js/dataexplorer.js]: Why?

File: molgenis-core-ui/src/main/resources/js/jquery.molgenis.tree.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 (function($, molgenis) {
 	"use strict";
 	
-	var restApi = new molgenis.RestClient();
+	var api = new molgenis.RestClient();
```
Comment (bartcharbon) [inline molgenis-core-ui/src/main/resources/js/jquery.molgenis.tree.js]: Why?

**Comment by fdlk** [comment]
I'll see if I can merge with master

File: molgenis-dataexplorer/src/main/resources/js/dataexplorer.js (hunk 1 above)
Comment (Mark-de-Haan) [inline molgenis-dataexplorer/src/main/resources/js/dataexplorer.js]: Ah its left over from the switch to v2 api and back again, il change it

**Comment by Mark-de-Haan** [comment]
@jjettenn  or @fdlk ??

**Comment by jjettenn** [comment]
@fdlk is assignee for the PR.

**Comment by jjettenn** [comment]
@fdlk shall I look at this PR and test it?

**Comment by fdlk** [comment]
@jjettenn sure!

**Pull request merged**